kafka/channel/pkg/dispatcher/dispatcher.go

```diff
@@ -318,11 +319,11 @@ func NewDispatcher(args *KafkaDispatcherArgs) (*KafkaDispatcher, error) {
 	return dispatcher, nil
 }
 
-func (d *KafkaDispatcher) getChannelReferenceFromHost(host string) (eventingchannels.ChannelReference, error) {
+func (d *KafkaDispatcher) getChannelReferenceFromHost(host url.URL) (eventingchannels.ChannelReference, error) {
 	chMap := d.getHostToChannelMap()
-	cr, ok := chMap[host]
+	cr, ok := chMap[host.Path]
 	if !ok {
-		return cr, fmt.Errorf("invalid Hostname:%s. Hostname not found in ConfigMap for any Channel", host)
+		return cr, fmt.Errorf("invalid Hostname: '%v'. Hostname not found in ConfigMap for any Channel", host)
```

Comment:
host is not longer just a host, it is a full uri so this comment and the variable name is out of date.

Reply:
What's the problem with the comment here? Do you prefer "invalid host" instead of "invalid hohstname"?

Reply:
host == a uri, not a hostname.

Reply:
URI is a type. "Host" and "Hostname" are just what we choose to call it (or not), so I don't quite understand your last comment. Can you please be more direct in what you'd like adjusted here? Do you prefer "invalid host" instead of "invalid hostname"?

Reply:
I just adjusted it to include host.Path instead of the full host.

Reply:
How about just renaming the variable?

Reply:
Sure, can do! I also adjusted it to be host.Path instead of the full host since this error message is right after an assertion on host.Path.